### PR TITLE
check if ebpf source file contain ' inline ' and should use __always_inline

### PR DESCRIPTION
**Pull request by nplanel**
### What does this PR do?

Sanity check if ebpf source file contain ' inline ' and we should use __always_inline

### Motivation

Inline macro is set/depend by the kernel header include/linux/compiler_types.h
Avoid failing CI a not inlining a function that generate bpf2bpf call

### CODE REVIEW

File: pkg/security/ebpf/c/pipe.h [hunk 1]
```diff
@@ -10,7 +10,7 @@ struct bpf_map_def SEC("maps/pipefs_mountid") pipefs_mountid = {
     .namespace = "",
 };
 
-#define DECLARE_EQUAL_TO_SUFFIXED(suffix, s) static inline int equal_to_##suffix(char *str) { \
+#define DECLARE_EQUAL_TO_SUFFIXED(suffix, s) static __attribute__((always_inline)) int equal_to_##suffix(char *str) { \
```
Comment (nplanel) [inline pkg/security/ebpf/c/pipe.h:13]: @lebauce are you agreed with this change ?

File: tasks/system_probe.py [hunk 2]
```diff
@@ -563,10 +563,15 @@ def get_ebpf_build_flags(target=None, kernel_release=None, minimal_kernel_releas
...
 
+def ebpf_check_source_file(ctx, src_file):
+    ctx.run("if grep -q ' inline ' {src_file}; then echo -e '\u001b[7mPlease use __always_inline instead of inline in {src_file}\u001b[0m';exit 1;fi".format(src_file=src_file), echo=False)
+
+
 def build_network_ebpf_compile_file(
     ctx, parallel_build, build_dir, p, debug, network_prebuilt_dir, network_flags, extension=".bc"
 ):
     src_file = os.path.join(network_prebuilt_dir, f"{p}.c")
+    ebpf_check_source_file(ctx, src_file)
```
Comment (brycekahle) [inline tasks/system_probe.py:574]: This only checks the `.c` file and not any of the `.h` files, correct?

Comment (nplanel) [reply tasks/system_probe.py:574]: yes this is correct
the runtime compilation include everything

Comment (brycekahle) [reply tasks/system_probe.py]: Runtime compilation uses different sources from prebuilt though. Should we just check all `.c` and `.h` files in specific directories? Perhaps at the start of the `build_object_files` function?

Comment (nplanel) [reply tasks/system_probe.py]: for runtime it's done in pkg/ebpf/bytecode/runtime/integrity.go
One would be better, but we don't have really one place of truth for source files

Comment (brycekahle) [reply tasks/system_probe.py]: I'd like to check more than just the top-level `.c` prebuilt files. Can we just scan all `.c` and `.h` files in ebpf source directories? That would cover both prebuilt and runtime.

**Review by brycekahle — CHANGES_REQUESTED** [review]
We need to scan more than just the top-level `.c` prebuilt files. We have a lot of functions in `.h` files that need this check.

File: tasks/system_probe.py (hunk 2 above)
Comment (nplanel) [reply tasks/system_probe.py]: done

**Review by brycekahle — APPROVED** [review]

**Review by paulcacheux — APPROVED** [review]

**Pull request merged**